### PR TITLE
[Bug] [API]After we delete the worker group, but this task group still keep in the task setting.

### LINKED ISSUE #12002: [Bug] [UI] After we delete the worker group, but this task group still keep in the task setting.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

in #12000, after deleted the worker group, then edit the task setting, the `app01` still show in the setting.
image

### What you expected to happen

the worker group should be null 

### How to reproduce

delete a worker group and then edit the task setting.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hstdream**
## Purpose of the pull request
close #12002 #11805 
After the workgroup is deleted, the workgroup in the workflow still exists.

## Brief change log

After the workgroup is deleted, the workgroup in the workflow still exists

## Verify this pull request

1.create worker group
2.create workflow select workergroup
3.delete worker group

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by hstdream** [comment]
@SbloodyS @caishunfeng  review

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessTaskRelationMapper.xml [hunk 1]
```diff
@@ -200,7 +200,11 @@
             and post_task_code = #{postTaskCode}
         </if>
     </select>
-
+    <select id="queryByWorkGroupName" resultType="org.apache.dolphinscheduler.dao.entity.ProcessTaskRelation">
+        select
+            relation.id
+        from t_ds_process_task_relation relation, t_ds_task_definition_log log where
+        log.code = relation.post_task_code and log.worker_group = #{workerGroupName}
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessTaskRelationMapper.xml:207]: It looks miss `</select>`, Why not query `t_ds_task_definition_log` directly?

**Review by caishunfeng — CHANGES_REQUESTED** [review]

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessTaskRelationMapper.xml (hunk 1 above)
Comment (caishunfeng) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessTaskRelationMapper.xml:207]: When user delete workerGroup, is it better to check references in `t_ds_task_definition`, but not `t_ds_task_definition_log`? 
If just the old version taskDefinition has this references, can it allow to delete?

cc @ruanwenjun @SbloodyS

**Comment by caishunfeng** [comment]
BTW, please add some UT for it, thanks.

**Comment by hstdream** [comment]
invalid pr.

**Pull request closed without merging**